examples/cpp/7_series/gnss_ins/7_series_gnss_ins_example.cpp

```diff
@@ -0,0 +1,1194 @@
+        gnssBaseRate1,
+        gnssDecimation1,
+        SAMPLE_RATE_HZ
+    );
+
+    // Descriptor rate is a pair of data descriptor set and decimation
+    const mip::DescriptorRate gnssDescriptors1[1] = {
+        { mip::data_gnss::FixInfo::FIELD_DESCRIPTOR, gnssDecimation1 }
+    };
+
+    MICROSTRAIN_LOG_INFO("Configuring message format for GNSS 1 data.\n");
+    cmdResult = mip::commands_3dm::writeMessageFormat(
+        _device,
+        mip::data_gnss::MIP_GNSS1_DATA_DESC_SET,                // Data descriptor set
+        sizeof(gnssDescriptors1) / sizeof(gnssDescriptors1[0]), // Number of descriptors to include
+        gnssDescriptors1                                        // Descriptor array
+    );
+
+    if (!cmdResult.isAck())
+    {
+        terminate(_device, cmdResult, "Could not set message format for GNSS 1 data!\n");
+    }
+
+    // Note: Typically, the base rates for all GNSS descriptors will be the same, but adding this for completeness
+    MICROSTRAIN_LOG_INFO("Getting the base rate for GNSS 2 data.\n");
+    uint16_t gnssBaseRate2;
+    cmdResult = mip::commands_3dm::getBaseRate(
+        _device,
+        mip::data_gnss::MIP_GNSS2_DATA_DESC_SET, // Data descriptor set
+        &gnssBaseRate2                           // Base rate out
+    );
+
+    if (!cmdResult.isAck())
+    {
+        terminate(_device, cmdResult, "Could not get GNSS 2 base rate!\n");
+    }
+
+    // Supported sample rates can be any value from 1 up to the base rate
+    // Note: Decimation can be anything from 1 to 65,565 (uint16_t::max)
+    if (SAMPLE_RATE_HZ == 0 || SAMPLE_RATE_HZ > gnssBaseRate2)
+    {
+        terminate(
+            _device,
+            mip::CmdResult::NACK_INVALID_PARAM,
+            "Invalid sample rate of %dHz! Supported rates are [1, %d].\n",
+            SAMPLE_RATE_HZ,
+            gnssBaseRate2
+        );
+    }
+
+    // Calculate the decimation (stream rate) for the device based on its base rate
+    const uint16_t gnssDecimation2 = gnssBaseRate2 / SAMPLE_RATE_HZ;
+    MICROSTRAIN_LOG_INFO("Decimating GNSS 2 base rate %d by %d to stream data at %dHz.\n",
+        gnssBaseRate2,
+        gnssDecimation2,
+        SAMPLE_RATE_HZ
+    );
+
+    // Descriptor rate is a pair of data descriptor set and decimation
+    const mip::DescriptorRate gnssDescriptors2[1] = {
+        { mip::data_gnss::FixInfo::FIELD_DESCRIPTOR, gnssDecimation2 }
+    };
+
+    MICROSTRAIN_LOG_INFO("Configuring message format for GNSS 2 data.\n");
+    cmdResult = mip::commands_3dm::writeMessageFormat(
+        _device,
+        mip::data_gnss::MIP_GNSS2_DATA_DESC_SET,                // Data descriptor set
+        sizeof(gnssDescriptors2) / sizeof(gnssDescriptors2[0]), // Number of descriptors to include
+        gnssDescriptors2                                        // Descriptor array
+    );
+
+    if (!cmdResult.isAck())
+    {
+        terminate(_device, cmdResult, "Could not set message format for GNSS 2 data!\n");
+    }
```

Comment:
Make this function configure only one receiver, then call it twice with appropriate parameters.